src/rgw/rgw_s3select_private.h

```diff
@@ -32,7 +32,14 @@
 #include <liboath/oath.h>
 
 
+#pragma GCC diagnostic push
+#pragma clang diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated"
+#pragma clang diagnostic ignored "-Wdeprecated"
```

Comment:
> Silence deprecation warnings from s3select

what is deprecated? s3select is our project, so i think it's probably better to resolve them? cc @galsalomon66

Reply:
@adamemerson 
please update me on these issues. 
I can do the necessary changes, in case it is required.

Reply:
> @adamemerson please update me on these issues. I can do the necessary changes, in case it is required.

The main issue I saw was lots of warnings that implicitly capturing `this` with `=` was deprecated.